appsec/src/extension/ddappsec.c

```diff
@@ -499,16 +498,28 @@ static PHP_FUNCTION(datadog_appsec_push_address)
         return;
     }
 
+    zend_array *parameters_arr =
+        zend_new_array(zend_hash_num_elements(addresses));
     zval parameters_zv;
-    zend_array *parameters_arr = zend_new_array(1);
     ZVAL_ARR(&parameters_zv, parameters_arr);
-    zend_hash_add(Z_ARRVAL(parameters_zv), key, value);
-    Z_TRY_ADDREF_P(value);
+
+    zend_string *key;
+    zval *value;
+    ZEND_HASH_FOREACH_STR_KEY_VAL(addresses, key, value)
```

Comment:
I'm not sure if a copy of the array is needed, as we discussed it might be better to deal with the potential lack of keys during serialisation to avoid a copy.

Reply:
Done